### DISCUSSION TITLE
Kind of wish you didn't included NodeJS stuff.

### DISCUSSION THREAD

**Discussion by jimmywarting** [oven-sh/bun#5518] [General]
Hey there Bun.js Team,

I've been following Bun.js closely, and I wanted to share some thoughts and feelings I have regarding the compatibility decisions with Node.js.

First off, kudos to the Bun.js team for the incredible work you've put into creating an innovative JavaScript runtime for web development. It's been an exciting journey watching Bun.js evolve.

That said, I have some reservations about the direction of making Bun.js compatible with Node.js. While I appreciate the intention to ease the transition for Node.js developers, I think Bun and Deno's both have fallen for the dominating NodeJS giant and feel like both needed to shipping every legacy piece of what NodeJS has instead of inovating and leading the technology forward with just only web standards. 

For me, Bun.js stood out because of its promise of embracing modern web technologies and b/c it use safaris fast engine. I'd personally prefer to see Bun.js focusing solely on web standards and its unique capabilities, pushing the boundaries of web development. The inclusion of Node.js-specific features like global process and buffer, will only make it so ppl continue using NodeJS stuff those making it more incompatible with Browsers runtimes, personally i think it's just better to only use things that works in every env. like using 
- Uint8Array instead of node:buffer 
- web streams instead of node:streams.
- direct-sockets instead of node:net
- whatwg/fs and file system access instead of node:fs
- web crypto instead of node:crypto
- web worker instead of node:worker_thread
- DecompressionStream & CompressionStream instead of node:zlib
- automatically generated import-map when installing packages and using them internally
- etc

In the rapidly evolving world of web development, I believe Bun.js has the potential to shine by sticking to web tech and promoting a more forward-looking approach. While it's crucial to support developers transitioning from Node.js, I'm curious to see if Bun.js could lead the way in a cleaner, more modern direction that embrace more web standards and cross functional programming

I probably know that you won't remove all NodeJS stuff. and i can understand if some ppl want it.
but enabeling developer to still use nodejs specific stuff will only make it so that ppl write NodeJS specific code that only works in some certain environment instead of coding something that works anywhere.

I kind of rather have wanted to see some NodeJS opt-in flag or something.

this is just some personal general toughs.
I don't expect any changes.
I don't want to be critical.
I like what you all have created.

Cheers,

**Comment by melroy89** [comment]
Fully agreed! Indeed it has overlap with my discussion: https://github.com/oven-sh/bun/discussions/5879

Tagging @Jarred-Sumner .

**Comment by Jarred-Sumner** [comment]
Bun is focused on productivity and many of Node’s choices are still more productive for developers than alternatives
